src/test/java/api/requests/scenarios/HoldShelfFulfillmentTests.java

```diff
@@ -0,0 +1,137 @@
+  }
+
+  @Test
+  public void checkInAtDifferentServicePointPlacesItemInTransit()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    final IndividualResource pickupServicePoint = servicePointsFixture.cd1();
+    final IndividualResource checkInServicePoint = servicePointsFixture.cd2();
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource james = usersFixture.james();
+    IndividualResource jessica = usersFixture.jessica();
+
+    loansFixture.checkOutByBarcode(smallAngryPlanet, james);
+
+    IndividualResource requestByJessica = requestsFixture.placeHoldShelfRequest(
+      smallAngryPlanet, jessica, DateTime.now(DateTimeZone.UTC),
+      pickupServicePoint.getId());
+
+    loansFixture.checkInByBarcode(smallAngryPlanet,
+      DateTime.now(DateTimeZone.UTC), checkInServicePoint.getId());
+
+    Response request = requestsClient.getById(requestByJessica.getId());
+
+    assertThat(request.getJson().getString("status"), is(OPEN_IN_TRANSIT));
+
+    smallAngryPlanet = itemsClient.get(smallAngryPlanet);
+
+    assertThat(smallAngryPlanet, hasItemStatus(IN_TRANSIT));
+
+    final String destinationServicePoint = smallAngryPlanet.getJson()
+      .getString("inTransitDestinationServicePointId");
+
+    assertThat("in transit item should have a destination",
+      destinationServicePoint, is(pickupServicePoint.getId()));
+  }
+
+  @Test
+  public void canBePickedUpWhenCheckedInAtPickupServicePointAfterTransit()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    final IndividualResource pickupServicePoint = servicePointsFixture.cd1();
+    final IndividualResource checkInServicePoint = servicePointsFixture.cd2();
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource james = usersFixture.james();
+    IndividualResource jessica = usersFixture.jessica();
+
+    loansFixture.checkOutByBarcode(smallAngryPlanet, james);
+
+    IndividualResource requestByJessica = requestsFixture.placeHoldShelfRequest(
+      smallAngryPlanet, jessica, DateTime.now(DateTimeZone.UTC),
+      pickupServicePoint.getId());
+
+    loansFixture.checkInByBarcode(smallAngryPlanet,
+      DateTime.now(DateTimeZone.UTC), checkInServicePoint.getId());
+
+    loansFixture.checkInByBarcode(smallAngryPlanet,
+      DateTime.now(DateTimeZone.UTC), pickupServicePoint.getId());
+
+    Response request = requestsClient.getById(requestByJessica.getId());
+
+    assertThat(request.getJson().getString("status"), is(OPEN_AWAITING_PICKUP));
+
+    smallAngryPlanet = itemsClient.get(smallAngryPlanet);
+
+    assertThat(smallAngryPlanet, hasItemStatus(AWAITING_PICKUP));
+
+    assertThat("awaiting pickup item should not have a destination",
+      smallAngryPlanet.getJson().containsKey("inTransitDestinationServicePointId"),
+      is(false));
+  }
```

Comment:
As in canBePickedUpWhenCheckedInAtPickupServicePoint() , should there be a step here to perform a checkout by Jessica to prove that it could be picked up?

Reply:
That's true, `can be` is ambiguous, it could either mean that it is ready to be picked up or that a patron can successfully pick it up.

I intended for those tests to cover the former. Have renamed them to try to reduce that ambiguity.